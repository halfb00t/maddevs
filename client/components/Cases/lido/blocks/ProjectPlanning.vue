<template>
  <section class="case-lido__project-planning m-96_bottom media-m-48_bottom">
    <div class="container_regular">
      <h3 class="case_title_h2 m-12_bottom">
        Project planning and management
      </h3>
      <TextParagraph class="m-24_bottom">
        To perform client tasks, the most qualified programmers were recruited.
        The project's first phase involved 4 people, while the second phase involved 3 developers.
      </TextParagraph>
      <TextParagraph class="m-24_bottom">
        Work on both objectives occurred in several phases:
      </TextParagraph>
      <List class="m-24_bottom">
        <ListItemCustomCheckMark
          v-for="listItem in projectPlanningList"
          :key="listItem.text"
          :background-color="listItem.backgroundColor"
          data-aos="fade-up"
          data-aos-anchor-placement="bottom"
          class="case-lido__story-behind-list-item"
        >
          <img
            slot="icon"
            v-lazy-load
            width="25"
            height="25"
            :data-src="$getMediaFromS3(`/images/Cases/lido/svg/${listItem.icon}.svg`)"
            :alt="listItem"
            class="list-item__box-icon"
          >
          <TextParagraph
            slot="description"
            :class="`case-lido__project-planning-list-item-description text-color-${listItem.textColor}`"
          >
            {{ listItem.text }}
          </TextParagraph>
        </ListItemCustomCheckMark>
      </List>
      <TextParagraph>
        *IMPORTANT: Lido ensures that external companies always audit smart contracts because it is a matter of security and the right functionality.
      </TextParagraph>
    </div>
  </section>
</template>
<script>
import TextParagraph from '@/components/Cases/shared/TextParagraph'
import ListItemCustomCheckMark from '@/components/Cases/shared/ListItemCustomCheckMark'
import List from '@/components/Cases/shared/List'
import { projectPlanningList } from '@/data/caseLido'

export default {
  name: 'ProjectPlanning',

  components: {
    TextParagraph,
    List,
    ListItemCustomCheckMark,
  },

  data() {
    return {
      projectPlanningList,
    }
  },
}
</script>
<style lang="scss" scoped>
.case-lido__project-planning {
  ::v-deep .list-item-custom-check-mark {
    padding: 18.92px 29.5px;
    margin-bottom: 16px;

    &:last-child {
      margin-bottom: 0;
    }

    @media screen and (max-width: 480px) {
      padding: 17.08px 13.5px;
    }
  }

  &-list-item-description {
    font-size: 20px;
    font-weight: 500;
    line-height: 166%;

    @media screen and (max-width: 480px) {
      font-size: 16px;
      line-height: 150%;
      letter-spacing: -0.02em;
    }
  }
}

.list-item__box-icon {
  vertical-align: text-bottom;
  margin-right: 27.44px;
  margin-top: 4px;

  @media screen and (max-width: 480px) {
    margin-right: 12.5px;
    margin-top: 0;
  }
}
</style>
