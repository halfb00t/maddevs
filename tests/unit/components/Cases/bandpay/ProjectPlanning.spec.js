import { render, screen } from '@testing-library/vue'
import { createLocalVue } from '@vue/test-utils'
import ProjectPlanning from '@/components/Cases/bandpay/ProjectPlanning'
import '../../../__mocks__/intersectionObserverMock'
// eslint-disable-next-line import/order
import lazyLoad from 'nuxt-lazy-load/lib/module'

const mocks = {
  $getMediaFromS3: () => 'img.jpg',
}

const localVue = createLocalVue()
localVue.directive('lazy-load', lazyLoad)
jest.mock('nuxt-lazy-load/lib/module')

describe('BandPay ProjectPlanning component', () => {
  it('should render correctly', () => {
    const { container } = render(ProjectPlanning, {
      mocks,
      localVue,
    })

    expect(container).toMatchSnapshot()
  })

  it('should render with text', () => {
    render(ProjectPlanning, {
      mocks,
      localVue,
    })

    expect(screen.getByText(/Project planning and management/i).className).toBeTruthy()
    expect(screen.getByText(/The size of the development team changed/i).className).toBeTruthy()
    expect(screen.getByText(/After the active development phase changed/i).className).toBeTruthy()
    expect(screen.getByText(/After the active development phase changed/i).className).toBeTruthy()
  })

  describe('KeyAchievements images', () => {
    it('should render images', () => {
      render(ProjectPlanning, {
        mocks,
        localVue,
      })

      screen.getAllByTestId('test-picture-img').forEach(item => {
        expect(item.getAttribute('data-src')).toBe(mocks.$getMediaFromS3())
      })
    })
  })
})
