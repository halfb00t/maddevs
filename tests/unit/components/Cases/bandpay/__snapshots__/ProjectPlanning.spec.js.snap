// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BandPay ProjectPlanning component should render correctly 1`] = `
<div>
  <section
    class="m-96_bottom media-m-48_bottom"
  >
    <section
      class="container_regular m-48_bottom media-m-24_bottom"
    >
      <h3
        class="case_title_h2 m-12_bottom"
      >
        
      Project planning and management
    
      </h3>
       
      <p
        class="case_paragraph m-24_bottom media-m-12_bottom"
      >
        
      The size of the development team changed throughout the whole project. The group was relatively small initially,
      but it scaled up sharply as we moved forward with the project and onboarded new team members, such as frontend,
      backend, and mobile developers. At that time, we used the Agile Scrum methodology, that is, divided the project
      into incremental phases or 2-week sprints and held regular demo and retrospective sessions.
    
      </p>
       
      <p
        class="case_paragraph m-24_bottom media-m-12_bottom"
      >
        
      After the active development phase changed for the support phase, the team size decreased, and we switched
      to Kanban. This methodology has no fixed sprints and timeframes, and it focuses more on the flow of the project.
      By applying Kanban, we prioritised tasks by pulling them into the workflow from the backlog.
    
      </p>
    </section>
     
    <section
      class="container_middle"
    >
      <div
        class="case_image-wrapper m-48_top media-m-24_top case_box-shadow"
      >
        <picture>
          <source
            class="image grey-background"
            data-srcset="img.jpg ,img.jpg"
            type="image/webp"
          />
           
          <img
            alt="Project Planning"
            class="image grey-background"
            data-src="img.jpg"
            data-srcset="img.jpg"
            data-testid="test-picture-img"
            height="16.45"
            width="1026"
          />
        </picture>
         
        <div
          class="container_middle__main-image"
        >
          <picture>
            <source
              class="image"
              data-srcset="img.jpg ,img.jpg"
              type="image/webp"
            />
             
            <img
              alt="Project Planning"
              class="image"
              data-src="img.jpg"
              data-srcset="img.jpg"
              data-testid="test-picture-img"
              height="474.71"
              width="944"
            />
          </picture>
        </div>
      </div>
    </section>
  </section>
</div>
`;
